app/models/conversion_host.rb

```diff
@@ -9,28 +9,109 @@ class ConversionHost < ApplicationRecord
 
   # To be eligible, a conversion host must have the following properties
   #  - A transport mechanism is configured for source (set by 3rd party)
-  #  - Credentials are set on the resource
+  #  - Credentials are set on the resource and SSH connection works
   #  - The number of concurrent tasks has not reached the limit
   def eligible?
-    source_transport_method.present? && check_resource_credentials && check_concurrent_tasks
+    source_transport_method.present? && check_ssh_connection && check_concurrent_tasks
```

Comment:
Where is this method going to be called from and how often?  Without storing or caching the result of the ssh_connection I'm worried this will be called a lot.

Could we use the existing authentication_check from the linked resource which stores the status in the authentication record?

Reply:
This will be called by the throttler to identify if hosts are eligible for a specific task. We could limit it to check that credential is present for the resource, and not that the connection is valid. This would move this check to the time we actually interact with the conversion_host: run conversion, role check/enable/disable...

`authentication_check` exists for Host but not for Vm. And in the case of Vm, the resource relies on credentials associated to the provider.

Reply:
Nope. I guess it was also a UI challenge as today we don't have authentication for Vm and that would mean integrating a new tab or config dialog in the UI. And if we add authentication to Vm, how does it interact with Embedded Ansible ? IMO, it should be shared with Embedded Ansible, as user would be frustrated to create authentication tokens in 2 different places.

So, it was decided that the private key used for migration would be attached to the OpenStack provider and that somehow the associated public key would be injected in the VM.

Reply:
For that one, do you want to limit the check to credentials existence ?